salt/_pillar/metalk8s_nodes.py

```diff
@@ -36,39 +26,43 @@ def node_info(node, ca_minion):
 
     roles = set()
 
-    if VERSION_LABEL in node.metadata.labels:
-        result['version'] = node.metadata.labels[VERSION_LABEL]
+    if VERSION_LABEL in node['metadata']['labels']:
+        result['version'] = node['metadata']['labels'][VERSION_LABEL]
 
-    for (label, value) in node.metadata.labels.items():
+    for (label, _) in node['metadata']['labels'].items():
         if label.startswith(ROLE_LABEL_PREFIX):
             role = label[len(ROLE_LABEL_PREFIX):]
             if role:
                 roles.add(role)
 
-    if node.metadata.name == ca_minion:
+    if node['metadata']['name'] == ca_minion:
         roles.add('ca')
 
     result['roles'] = list(roles)
 
     return result
 
 
-def get_cluster_version(api_client):
-    coreV1 = kubernetes.client.CoreV1Api(api_client=api_client)
+def get_cluster_version(kubeconfig=None):
     try:
-        namespace = coreV1.read_namespace(name="kube-system")
-    except (ApiException, HTTPError) as exc:
+        namespace = __salt__['metalk8s_kubernetes.get_object'](
+            name="kube-system",
+            kind="Namespace",
+            apiVersion="v1",
+            kubeconfig=kubeconfig
+        )
+    except Exception as exc:  # pylint: disable=broad-except
```

Comment:
It's an execution module, so I guess it should only raise a `CommandExecutionError`?